### ISSUE TITLE
Config representation is broken

### ISSUE THREAD

**Issue by annawoodard** [Parsl/parsl#916]
The config representation appears to be broken:
```
In [1]: from parsl.configs.htex_local import config

In [1]: config
Out[2]: Config()
```

Git bisect tells me the last commit where this error didn't happen was c6deae01a2840b8617c0b3975c9f8c99882a0c60. The commits immediately after that I haven't tested because I'm getting some error from the `enforce` package. Then by 695d2763449026f9779a61a2293960ea531d89bb I see the error. So this appears to have something to do with the type checking.

cc'ing @benclifford

**Comment by annawoodard** [comment]
Removing the `typechecked` decorator [here](https://github.com/Parsl/parsl/blob/master/parsl/config.py#L55) fixes the representation. We'll need to poke around to figure out how to make them play nicely.

**Comment by benclifford** [comment]
`RepresentationMixin` gets the list of parameters to display by inspecting the argument list of `__init__` and expecting that argument list to explicitly list all the relevant arguments:

```
>>> inspect.getfullargspec(Config.__init__)
FullArgSpec(args=['self', 'executors', 'app_cache', 'checkpoint_files', 'checkpoint_mode', 'checkpoint_period', 'data_management_max_threads', 'lazy_errors', 'retries', 'run_dir', 'strategy', 'monitoring', 'usage_tracking'], varargs=None, varkw=None, defaults=(None, True, None, None, None, 10, True, 0, 'runinfo', 'simple', None, False), kwonlyargs=[], kwonlydefaults=None, annotations={})
```

The `@typechecked` decorator replaces that `__init__` with a different init that performs its own type checking and then passes on every keyword arg to the original `__init__`  (so doing similar to RepresentationMixin). That replacement `__init__` does not list the arguments explicitly:

```
>>> inspect.getfullargspec(Config.__init__)
FullArgSpec(args=[], varargs='args', varkw='kwargs', defaults=None, kwonlyargs=[], kwonlydefaults=None, annotations={})
```

**Comment by benclifford** [comment]
This specific problem is fixed by #927 which modifies RepresentationMixin to look for a single layer of `functools.update_wrapper` wrapping - which is used by the @typechecked decorator. #927 does not make RepresentationMixin cope with arbitrary decorators, though.
